src/Tasks.UnitTests/ResourceHandling/GenerateResource_Tests.cs

```diff
@@ -382,27 +384,29 @@ public void ForceOutOfDateLinked()
 
                 Utilities.ExecuteTask(t);
 
-                string resourcesFile = t.OutputResources[0].ItemSpec;
-                Assert.Equal(".resources", Path.GetExtension(resourcesFile));
-                resourcesFile = t.FilesWritten[0].ItemSpec;
-                Assert.Equal(".resources", Path.GetExtension(resourcesFile));
-                
+                Path.GetExtension(t.OutputResources[0].ItemSpec).ShouldBe(".resources");
+                Path.GetExtension(t.FilesWritten[0].ItemSpec).ShouldBe(".resources");
+
                 Utilities.AssertStateFileWasWritten(t);
 
-                GenerateResource t2 = Utilities.CreateTask(_output);
+                GenerateResource t2 = Utilities.CreateTask(_output, usePreserialized, _env);
                 t2.StateFile = new TaskItem(t.StateFile);
                 t2.Sources = new ITaskItem[] { new TaskItem(resxFile) };
 
-                DateTime time = File.GetLastWriteTime(t.OutputResources[0].ItemSpec);
+                DateTime firstWriteTime = File.GetLastWriteTime(t.OutputResources[0].ItemSpec);
                 System.Threading.Thread.Sleep(200);
-                File.SetLastWriteTime(bitmap, DateTime.Now);
+                File.SetLastWriteTime(bitmap, DateTime.Now + TimeSpan.FromSeconds(2));
 
                 Utilities.ExecuteTask(t2);
 
-                Assert.True(DateTime.Compare(File.GetLastWriteTime(t2.OutputResources[0].ItemSpec), time) > 0);
+                File.GetLastWriteTime(t2.OutputResources[0].ItemSpec).ShouldBeGreaterThan(firstWriteTime);
```

Comment:
Is there a test that asserts the task is incremental if you run t2 without updating the timestamp of the bitmap?

Reply:
Ah, yes, there is :)